### PR TITLE
Add mdoc + docusaurus build to repo and run in CI

### LINKED ISSUE #3311: Migrate mdoc build to this repository and run it in CI
Take the mdoc + docusaurus build in https://github.com/geotrellis/geotrellis-docs, update all dependencies, and add it to this repo. 

When CI runs, mdoc should compile and build all doc pages and the docusaurus site should be built and published to some as yet unknown URL.

### PR DESCRIPTION
**Pull request by CloudNiner**
# Overview

This PR adds the skeleton for an mdoc + docusaurus static documentation in the same vein as a number of other open source Scala projects. Never again will our doc examples be out of date 🎉. No JS or CSS knowledge is necessary to write docs.

The general idea here is to:
1. Start an mdoc watcher via sbt which listens for changes in `./docs-mdoc`
2. The mdoc watcher writes to `website/docs`
3. A dev server running in `./website` picks up changes in `./website/docs` and rerenders the docs site
4. Devs can visit http://localhost:3000 to see the docs site

More detailed instructions for these steps are in `website/README.md`, follow those to test this PR to double check correctness.

## Checklist

- [ ] ~~[./CHANGELOG.md](https://github.com/locationtech/geotrellis/blob/master/CHANGELOG.md) updated, if necessary. Link to the issue if closed, otherwise the PR.~~
- [ ] ~~[Module Hierarchy](https://github.com/locationtech/geotrellis/blob/master/docs/guide/module-hierarchy.rst) updated, if necessary~~
- [ ] ~~`docs` guides update, if necessary~~
- [ ] ~~New user API has useful Scaladoc strings~~
- [ ] ~~Unit tests added for bug-fix or new feature~~

## Demo

Is that output from a RasterSource call right there on a docs website? You better bet your bottom dollar it is.

Screen Shot 2020-12-10 at 5 53 59 PM
Screen Shot 2020-12-10 at 5 54 10 PM

## Notes

I didn't make any changes to the changelog, because there's no user facing changes yet. This is just a skeleton. Future issues will begin building out content in the `./docs-mdoc` directory.

Most of the files in `./website` are autogenerated by the docusaurus generator. I made changes to the config and sidebars files, and updated the favicon and header icon. 

Closes #3311

### CODE REVIEW

File: website/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+Ensure you have Node 12+ installed.
+
+In one terminal, run `./sbt "project mdoc;mdoc --watch"` in the parent directory.
+This will run the mdoc watcher which recompiles docs in `../docs-mdoc` and writes the output to `./docs`.
+
+Start the docusaurus dev server with:
+
+```shell
+yarn start
```
Comment (echeipesh) [inline website/README.md:15]: common knowledge but may be worth mentioning running 'yarn install' the first time?

File: website/docs/emr-deployments.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+These parameters above can be also set / overrided via the `spark-submit --conf` settings.
+
+### Configuration for SBT Lighter Plugin
+
+Before looking into this section, you may be interested in the [GeoTrellis Spark Job](https://github.com/geotrellis/geotrellis-spark-job.g8) template project as well.
+
+The SBT Lighter Plugin configuration example:
+
+```scala
```
Comment (echeipesh) [inline website/docs/emr-deployments.md:55]: This doesn't seem to give syntax highlight for the block like https://geotrellis.github.io/geotrellis-workshop/docs/layer-model does, guessing this is dependent on the theme?

Screen Shot 2020-12-16 at 9 06 42 AM

**Review by echeipesh — APPROVED** [review]
I was able to:
- build the docs
- start the site
- add a new item in the side bar
- fail the build when I broke a code block in the docs

I think this is good enough to start writing the docs.
Currently the theme does not provide scala syntax highlighting.
That's not a blocker for this PR I'd say but should be fixed soon.

File: website/docs/emr-deployments.md (hunk 2 above)
Comment (CloudNiner) [reply website/docs/emr-deployments.md]: Fixed in https://github.com/locationtech/geotrellis/pull/3317/commits/f0e24a42a5d5afda556e411cde06b431cb7488a9

Screen Shot 2020-12-16 at 10 33 15 AM

File: website/README.md (hunk 1 above)
Comment (CloudNiner) [reply website/README.md:21]: Good call, perhaps not for mostly Scala folks, updated.

**Comment by CloudNiner** [comment]
Thanks for the review 🎉 will merge once build is green.

**Pull request merged**